### ISSUE TITLE
httpChallenge onHostRule doesn't respect acme.entryPoint

### ISSUE THREAD

**Issue by OlivierCuyp** [containous/traefik#3918]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I started traefik in a Swarm cluster of 6 AWS EC2 (3 master + 3 workers).
I defined 3 entrypoints http (:80) , https (:443), debug (:10000).
I started 2 services:

- test1 with http & https (host: test1.my-domain.pro)
- test2 with debug only (host: stack-test2.my-domain.pro)

### What did you expect to see?

I expected to only see certificate generation for **test1.my-domain.pro** but NOT for **stack-test2.my-domain.pro**.

### What did you see instead?

I saw acme certificate generation logs for **stack-test2.my-domain.pro** (see log section)

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.7.0-rc5
Codename:     maroilles
Go version:   go1.11
Built:        2018-09-18_01:50:00PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

**traefik.toml**
```toml
defaultEntryPoints = ["http", "https"]
logLevel = "INFO"

[api]
  dashboard = true
[entryPoints]
  [entryPoints.debug]
  address = ":10000"
  compress = true
  [entryPoints.http]
  address = ":80"
  compress = true
  [entryPoints.https]
  address = ":443"
  compress = true

[acme]
email="info@my-domain.com"
acmelogging=true
storage="/etc/traefik/acme/acme.json"
entryPoint = "https"
onHostRule = true
onDemand = true
  [acme.httpChallenge]
  entryPoint = "http"

[docker]
swarmMode = true
domain = "my-domain.pro"
watch = true
```

**stack.yml**
```yml
services:
  test1:
    image: company/service1:v123
    networks:
      - frontend
    deploy:
      mode: replicated
      replicas: 2
      resources:
        limits:
          memory: 256M
      labels:
        traefik.backend.loadbalancer.swarm: "false"
        traefik.frontend.rule: "Host:test1.my-domain.pro"
        traefik.frontend.passHostHeader: "true"
        traefik.frontend.entryPoints: "http,https"
        traefik.port: "80"
        traefik.docker.network: "frontend"
  test2:
    image: company/service2:v123
    networks:
      - backend
    deploy:
      mode: replicated
      replicas: 2
      resources:
        limits:
          memory: 256M
      labels:
        traefik.backend.loadbalancer.swarm: "false"
        traefik.frontend.passHostHeader: "true"
        traefik.frontend.entryPoints: "debug"
        traefik.port: "80"
        traefik.docker.network: "backend"

networks:
  frontend:
    external: true
  backend:
    external: true
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
time="2018-09-22T20:17:11Z" level=error msg="Unable to obtain ACME certificate for domains \"test2.my-domain.pro\" detected thanks to rule \"Host:stack-test2.my-domain.pro\" : unable to generate a certificate for the domains [stack-test2.my-domain.pro]: acme: Error -> One or more domains had a problem:\n[stack-test2.my-domain.pro] acme: Error 403 - urn:ietf:params:acme:error:unauthorized - Invalid response from http://stack-test2.my-domain.pro/.well-known/acme-challenge/sGKdvFnQUZPR7Ai3JDqG5Wa0T6se31mbu0CUBjEfR88 [34.234.134.165]: 503\n"
```

**Comment by OlivierCuyp** [comment]
Just a bit more background:

We are in a "web agency" where all the projects are micro-services based.
The security group opens only the ports 80 & 443 on the public interface.
When on the private interface all ports are open.
Only the `<project>-api` &  `<project>-front` are bind to **http/https** entryPoints.
While all the `other micro-services` are only bind to the **debug** entryPoint.
We need this so we can access to internal micro services (on port 10000) through the VPN on their private interface for direct debugging.

**Comment by traefiker** [comment]
Closed by #3923.
